# Exercise 4.8

# Table of Contents – Exercise 4.8
- 01 Import Libraries
- 02- Pathways
- 03– Groupby/Agg Mean
- 04– Customer Loyalty Flags
- 05– Price Analysis by Loyalty Segment
- 06– Classifying Users by Spending Behavior
- 07– Frequent vs Non-Frequent Customers
- 08- Work on Final Report
- 09- Export
- 10- Rebuilding Dept #/Name list with Product Order Counts

## 01 Import Libraries

In [1]:
# Libraries
import pandas as pd
import numpy as np
import os

## 02 Pathways

In [2]:
# File pathway shortcut
path = r'C:\Users\Chase\anaconda_projects\Exercise 4\07-2025 Instacart Basket Analysis'

In [3]:
# Prepared file - Orders_Products_Merge (Clean)
df_ords_prods_merge = pd.read_pickle(os.path.join(path, '02 Data', 'Prepared Data', 'ords_prods_merge_v1.pkl'))

EOFError: Ran out of input

In [ ]:
# Prepared file - Orders_Products_Merge (Clean)
df_ords_prods_merge = pd.read_pickle(os.path.join(path, '02 Data', 'Prepared Data', 'ords_prods_merge.pkl'))

In [44]:
# Prepared file - Orders_Products_Merge (Clean)
df_ords_prods_merge = pd.read_pickle(os.path.join(path, '02 Data', 'Prepared Data', 'ords_prods_merge_temp.pkl'))

In [4]:
df_ords_prods_merge.shape

(32435059, 18)

In [5]:
print(df_ords_prods_merge.columns.tolist())

['order_id', 'product_id', 'add_to_cart_order', 'reordered', 'user_id', 'order_number', 'orders_day_of_week', 'order_hour_of_day', 'days_since_prior_order', 'first_order', 'product_name', 'aisle_id', 'department_id', 'prices', '_merge', 'price_range', 'busiest_day', 'busiest_period_of_day']


In [7]:
df_ords_prods_merge.head()

order_id  product_id  add_to_cart_order  reordered  user_id  order_number  \
0         2       33120                  1          1   202279             3   
1         2       28985                  2          1   202279             3   
2         2        9327                  3          0   202279             3   
3         2       45918                  4          1   202279             3   
4         2       30035                  5          0   202279             3   

   orders_day_of_week  order_hour_of_day  days_since_prior_order  first_order  \
0                   5                  9                     8.0        False   
1                   5                  9                     8.0        False   
2                   5                  9                     8.0        False   
3                   5                  9                     8.0        False   
4                   5                  9                     8.0        False   

            product_name  aisle_id  department_id  prices _merge  \
0     Organic Egg Whites      86.0           16.0    11.3   both   
1  Michigan Organic Kale      83.0            4.0    13.4   both   
2          Garlic Powder     104.0           13.0     3.6   both   
3         Coconut Butter      19.0           13.0     8.4   both   
4      Natural Sweetener      17.0           13.0    13.7   both   

         price_range     busiest_day busiest_period_of_day  
0  Mid-range product  Regularly busy           Most orders  
1  Mid-range product  Regularly busy           Most orders  
2  Low-range product  Regularly busy           Most orders  
3  Mid-range product  Regularly busy           Most orders  
4  Mid-range product  Regularly busy           Most orders

In [8]:
print(df_ords_prods_merge.columns.tolist())

['order_id', 'product_id', 'add_to_cart_order', 'reordered', 'user_id', 'order_number', 'orders_day_of_week', 'order_hour_of_day', 'days_since_prior_order', 'first_order', 'product_name', 'aisle_id', 'department_id', 'prices', '_merge', 'price_range', 'busiest_day', 'busiest_period_of_day']


## 03 - Groupby/Agg Mean

Agg() mean of order_number grouped by department_id (full dataset)

In [9]:
df_ords_prods_merge.head()

order_id  product_id  add_to_cart_order  reordered  user_id  order_number  \
0         2       33120                  1          1   202279             3   
1         2       28985                  2          1   202279             3   
2         2        9327                  3          0   202279             3   
3         2       45918                  4          1   202279             3   
4         2       30035                  5          0   202279             3   

   orders_day_of_week  order_hour_of_day  days_since_prior_order  first_order  \
0                   5                  9                     8.0        False   
1                   5                  9                     8.0        False   
2                   5                  9                     8.0        False   
3                   5                  9                     8.0        False   
4                   5                  9                     8.0        False   

            product_name  aisle_id  department_id  prices _merge  \
0     Organic Egg Whites      86.0           16.0    11.3   both   
1  Michigan Organic Kale      83.0            4.0    13.4   both   
2          Garlic Powder     104.0           13.0     3.6   both   
3         Coconut Butter      19.0           13.0     8.4   both   
4      Natural Sweetener      17.0           13.0    13.7   both   

         price_range     busiest_day busiest_period_of_day  
0  Mid-range product  Regularly busy           Most orders  
1  Mid-range product  Regularly busy           Most orders  
2  Low-range product  Regularly busy           Most orders  
3  Mid-range product  Regularly busy           Most orders  
4  Mid-range product  Regularly busy           Most orders

In [10]:
# Aggregated mean of order_number grouped by department_id
df_ords_prods_merge.groupby('department_id').agg({'order_number': ['mean']})

order_number
                      mean
department_id             
1.0              15.457838
2.0              17.277920
3.0              17.170395
4.0              17.811403
5.0              15.215751
6.0              16.439806
7.0              17.225802
8.0              15.340650
9.0              15.895474
10.0             20.197148
11.0             16.170638
12.0             15.887671
13.0             16.583536
14.0             16.773669
15.0             16.165037
16.0             17.665606
17.0             15.694469
18.0             19.310397
19.0             17.177343
20.0             16.473447
21.0             22.902379

Question 3: Comparing the 1M subset to the entire dataset doesn't vary very much. Some results are only off by 1 or 2 and most aer the same result

## 04 - Customer Loyalty Flags

In [11]:
df_ords_prods_merge.groupby('department_id').agg({'order_number': ['mean', 'min', 'max']})

order_number        
                      mean min max
department_id                     
1.0              15.457838   1  99
2.0              17.277920   1  99
3.0              17.170395   1  99
4.0              17.811403   1  99
5.0              15.215751   1  99
6.0              16.439806   1  99
7.0              17.225802   1  99
8.0              15.340650   1  99
9.0              15.895474   1  99
10.0             20.197148   1  99
11.0             16.170638   1  99
12.0             15.887671   1  99
13.0             16.583536   1  99
14.0             16.773669   1  99
15.0             16.165037   1  99
16.0             17.665606   1  99
17.0             15.694469   1  99
18.0             19.310397   1  99
19.0             17.177343   1  99
20.0             16.473447   1  99
21.0             22.902379   1  99

In [12]:
# 3 step process to create a max column: groupby/transform/create
df_ords_prods_merge['max_order'] = df_ords_prods_merge.groupby(['user_id'])['order_number'].transform(np.max)

C:\Users\Chase\AppData\Local\Temp\ipykernel_17384\3404093928.py:2: FutureWarning: The provided callable <function max at 0x0000026308BEAAC0> is currently using SeriesGroupBy.max. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "max" instead.
  df_ords_prods_merge['max_order'] = df_ords_prods_merge.groupby(['user_id'])['order_number'].transform(np.max)


In [13]:
# Check work, ensure max_order column was created
df_ords_prods_merge.head(15)

order_id  product_id  add_to_cart_order  reordered  user_id  order_number  \
0          2       33120                  1          1   202279             3   
1          2       28985                  2          1   202279             3   
2          2        9327                  3          0   202279             3   
3          2       45918                  4          1   202279             3   
4          2       30035                  5          0   202279             3   
5          2       17794                  6          1   202279             3   
6          2       40141                  7          1   202279             3   
7          2        1819                  8          1   202279             3   
8          2       43668                  9          0   202279             3   
9          3       33754                  1          1   205970            16   
10         3       24838                  2          1   205970            16   
11         3       17704                  3          1   205970            16   
12         3       21903                  4          1   205970            16   
13         3       17668                  5          1   205970            16   
14         3       46667                  6          1   205970            16   

    orders_day_of_week  order_hour_of_day  days_since_prior_order  \
0                    5                  9                     8.0   
1                    5                  9                     8.0   
2                    5                  9                     8.0   
3                    5                  9                     8.0   
4                    5                  9                     8.0   
5                    5                  9                     8.0   
6                    5                  9                     8.0   
7                    5                  9                     8.0   
8                    5                  9                     8.0   
9                    5                 17                    12.0   
10                   5                 17                    12.0   
11                   5                 17                    12.0   
12                   5                 17                    12.0   
13                   5                 17                    12.0   
14                   5                 17                    12.0   

    first_order                                       product_name  aisle_id  \
0         False                                 Organic Egg Whites      86.0   
1         False                              Michigan Organic Kale      83.0   
2         False                                      Garlic Powder     104.0   
3         False                                     Coconut Butter      19.0   
4         False                                  Natural Sweetener      17.0   
5         False                                            Carrots      83.0   
6         False                   Original Unflavored Gelatine Mix     105.0   
7         False           All Natural No Stir Creamy Almond Butter      88.0   
8         False                            Classic Blend Cole Slaw     123.0   
9         False  Total 2% with Strawberry Lowfat Greek Strained...     120.0   
10        False                             Unsweetened Almondmilk      91.0   
11        False                                             Lemons     123.0   
12        False                               Organic Baby Spinach     123.0   
13        False    Unsweetened Chocolate Almond Breeze Almond Milk      91.0   
14        False                                Organic Ginger Root      83.0   

    department_id  prices _merge        price_range     busiest_day  \
0            16.0    11.3   both  Mid-range product  Regularly busy   
1             4.0    13.4   both  Mid-range product  Regularly busy   
2            13.0     3.6   both  Low-range product  Regularly busy   
3            13.0     8.4   both  Mid

Max Column created

In [14]:
# Assign loyalty flags based on max_order values --- Loyal Cust.
df_ords_prods_merge.loc[df_ords_prods_merge['max_order'] > 40, 'loyalty_flag'] = 'Loyal customer'

In [15]:
# Assign loyalty flags based on max_order values --- Reg. Cust.
df_ords_prods_merge.loc[(df_ords_prods_merge['max_order'] <= 40) & (df_ords_prods_merge['max_order'] > 10), 'loyalty_flag'] = 'Regular customer'

In [16]:
# Assign loyalty flags based on max_order values --- New Cust.
df_ords_prods_merge.loc[df_ords_prods_merge['max_order'] <= 10, 'loyalty_flag'] = 'New customer'

In [17]:
# Check work, ensure Flags worked
df_ords_prods_merge.head(15)

order_id  product_id  add_to_cart_order  reordered  user_id  order_number  \
0          2       33120                  1          1   202279             3   
1          2       28985                  2          1   202279             3   
2          2        9327                  3          0   202279             3   
3          2       45918                  4          1   202279             3   
4          2       30035                  5          0   202279             3   
5          2       17794                  6          1   202279             3   
6          2       40141                  7          1   202279             3   
7          2        1819                  8          1   202279             3   
8          2       43668                  9          0   202279             3   
9          3       33754                  1          1   205970            16   
10         3       24838                  2          1   205970            16   
11         3       17704                  3          1   205970            16   
12         3       21903                  4          1   205970            16   
13         3       17668                  5          1   205970            16   
14         3       46667                  6          1   205970            16   

    orders_day_of_week  order_hour_of_day  days_since_prior_order  \
0                    5                  9                     8.0   
1                    5                  9                     8.0   
2                    5                  9                     8.0   
3                    5                  9                     8.0   
4                    5                  9                     8.0   
5                    5                  9                     8.0   
6                    5                  9                     8.0   
7                    5                  9                     8.0   
8                    5                  9                     8.0   
9                    5                 17                    12.0   
10                   5                 17                    12.0   
11                   5                 17                    12.0   
12                   5                 17                    12.0   
13                   5                 17                    12.0   
14                   5                 17                    12.0   

    first_order                                       product_name  aisle_id  \
0         False                                 Organic Egg Whites      86.0   
1         False                              Michigan Organic Kale      83.0   
2         False                                      Garlic Powder     104.0   
3         False                                     Coconut Butter      19.0   
4         False                                  Natural Sweetener      17.0   
5         False                                            Carrots      83.0   
6         False                   Original Unflavored Gelatine Mix     105.0   
7         False           All Natural No Stir Creamy Almond Butter      88.0   
8         False                            Classic Blend Cole Slaw     123.0   
9         False  Total 2% with Strawberry Lowfat Greek Strained...     120.0   
10        False                             Unsweetened Almondmilk      91.0   
11        False                                             Lemons     123.0   
12        False                               Organic Baby Spinach     123.0   
13        False    Unsweetened Chocolate Almond Breeze Almond Milk      91.0   
14        False                                Organic Ginger Root      83.0   

    department_id  prices _merge        price_range     busiest_day  \
0            16.0    11.3   both  Mid-range product  Regularly busy   
1             4.0    13.4   both  Mid-range product  Regularly busy   
2            13.0     3.6   both  Low-range product  Regularly busy   
3            13.0     8.4   both  Mid

Flags look good

## 05 - Price Analysis by Loyalty Segment

In [18]:
# Check order standings per group
df_ords_prods_merge['loyalty_flag'].value_counts()

loyalty_flag
Regular customer    15891507
Loyal customer      10294027
New customer         6249525
Name: count, dtype: int64

In [19]:
# Descrpitive data on prices
price_stats = df_ords_prods_merge.groupby('loyalty_flag')['prices'].describe().T

In [20]:
# Work check
print(price_stats)

loyalty_flag  Loyal customer  New customer  Regular customer
count           1.028409e+07  6.243990e+06      1.587678e+07
mean            7.773468e+00  7.801099e+00      7.798158e+00
std             4.205895e+00  4.266466e+00      4.254553e+00
min             1.000000e+00  1.000000e+00      1.000000e+00
25%             4.200000e+00  4.200000e+00      4.200000e+00
50%             7.400000e+00  7.400000e+00      7.400000e+00
75%             1.120000e+01  1.130000e+01      1.130000e+01
max             2.500000e+01  2.500000e+01      2.500000e+01


Looking at the average and median prices are nearly identical for loyal, regular, and new customers (mean about 7.77–7.80, median = 7.40).
Std and max prices are also similar, indicating consistent spending behavior across groups

## 06 - Classifying Users by Spending Behavior

In [21]:
# Find user AVG price per product
user_avg_price = df_ords_prods_merge.groupby('user_id')['prices'].mean().reset_index()

In [22]:
# New Avg Price column to keep things clean
user_avg_price['avg_price'] = user_avg_price['prices']

In [23]:
# work check
user_avg_price.head()

user_id    prices  avg_price
0        1  6.367797   6.367797
1        2  7.515897   7.515897
2        3  8.197727   8.197727
3        4  8.205556   8.205556
4        5  9.189189   9.189189

In [24]:
# Function to create spending flags per customer
def spending_category(x):
    if x < 10:
        return 'Low spender'
    else:
        return 'High spender'

In [25]:
# Create a new spending flags column and apply spending flags to customers
user_avg_price['spending_flag'] = user_avg_price['avg_price'].apply(spending_category)

In [26]:
# Work check
user_avg_price.head()

user_id    prices  avg_price spending_flag
0        1  6.367797   6.367797   Low spender
1        2  7.515897   7.515897   Low spender
2        3  8.197727   8.197727   Low spender
3        4  8.205556   8.205556   Low spender
4        5  9.189189   9.189189   Low spender

## 07 - Frequent vs Non Frequent Customers

In [27]:
# Group by user_id and calculate the median days between orders
user_medians = df_ords_prods_merge.groupby('user_id')['days_since_prior_order'].median().reset_index()

In [28]:
# Keep track of user medians
user_medians.rename(columns={'days_since_prior_order': 'median_days'}, inplace=True)

In [29]:
user_medians.head()

user_id  median_days
0        1         20.0
1        2         13.0
2        3          9.0
3        4         17.0
4        5         11.0

Looks like the column was created and renamed correctly

In [30]:
# Define order freq. flag function
def order_frequency_flag(median):
    if median > 20:
        return 'Non-frequent customer'
    elif 10 <= median <= 20:
        return 'Regular customer'
    else:
        return 'Frequent customer'

In [31]:
# Apply function
user_medians['order_frequency_flag'] = user_medians['median_days'].apply(order_frequency_flag)

In [32]:
# Work check
user_medians.head()

user_id  median_days order_frequency_flag
0        1         20.0     Regular customer
1        2         13.0     Regular customer
2        3          9.0    Frequent customer
3        4         17.0     Regular customer
4        5         11.0     Regular customer

In [33]:
# Merge this work to the main dataset
df_ords_prods_merge = df_ords_prods_merge.merge(
    user_medians[['user_id', 'order_frequency_flag']],
    on='user_id',
    how='left'
)

In [34]:
# Merge and work check
df_ords_prods_merge.head()

order_id  product_id  add_to_cart_order  reordered  user_id  order_number  \
0         2       33120                  1          1   202279             3   
1         2       28985                  2          1   202279             3   
2         2        9327                  3          0   202279             3   
3         2       45918                  4          1   202279             3   
4         2       30035                  5          0   202279             3   

   orders_day_of_week  order_hour_of_day  days_since_prior_order  first_order  \
0                   5                  9                     8.0        False   
1                   5                  9                     8.0        False   
2                   5                  9                     8.0        False   
3                   5                  9                     8.0        False   
4                   5                  9                     8.0        False   

   ... aisle_id  department_id  prices  _merge        price_range  \
0  ...     86.0           16.0    11.3    both  Mid-range product   
1  ...     83.0            4.0    13.4    both  Mid-range product   
2  ...    104.0           13.0     3.6    both  Low-range product   
3  ...     19.0           13.0     8.4    both  Mid-range product   
4  ...     17.0           13.0    13.7    both  Mid-range product   

      busiest_day busiest_period_of_day max_order  loyalty_flag  \
0  Regularly busy           Most orders         8  New customer   
1  Regularly busy           Most orders         8  New customer   
2  Regularly busy           Most orders         8  New customer   
3  Regularly busy           Most orders         8  New customer   
4  Regularly busy           Most orders         8  New customer   

    order_frequency_flag  
0  Non-frequent customer  
1  Non-frequent customer  
2  Non-frequent customer  
3  Non-frequent customer  
4  Non-frequent customer  

[5 rows x 21 columns]

Looks like the merge was successful

In [35]:
# Overwrite original file with enriched version
df_ords_prods_merge.to_pickle(os.path.join(path, '02 Data', 'Prepared Data', 'cust_loyalty_ords_prods_merge.pkl'))

# 08 Work on Final Report

#### Price Range

In [36]:
# Define price range labeling function
def price_label(row):
    if row['prices'] <= 5:
        return 'Low-range product'
    elif 5 < row['prices'] <= 15:
        return 'Mid-range product'
    elif row['prices'] > 15:
        return 'High-range product'
    else:
        return 'Not enough data'

In [37]:
# Apply function to full dataset
df_ords_prods_merge['price_range'] = df_ords_prods_merge.apply(price_label, axis=1)

In [38]:
# Quick preview
print(df_ords_prods_merge[['prices', 'price_range']].head(10))

   prices        price_range
0    11.3  Mid-range product
1    13.4  Mid-range product
2     3.6  Low-range product
3     8.4  Mid-range product
4    13.7  Mid-range product
5    10.7  Mid-range product
6    11.5  Mid-range product
7    11.5  Mid-range product
8     7.5  Mid-range product
9    11.8  Mid-range product


In [39]:
# Count of each price range
price_counts = df_ords_prods_merge['price_range'].value_counts(dropna=False)

In [40]:
# Percentage of each price range
price_percent = df_ords_prods_merge['price_range'].value_counts(normalize=True, dropna=False) * 100

In [41]:
# Combine into summary DataFrame
price_summary = pd.DataFrame({
    'Price Range': price_counts.index,
    'Count': price_counts.values,
    '% of Total Orders': price_percent.round(1).astype(str) + '%'
})

In [42]:
# Display summary
print(price_summary)

                           Price Range     Count % of Total Orders
price_range                                                       
Mid-range product    Mid-range product  21865987             67.4%
Low-range product    Low-range product  10126321             31.2%
High-range product  High-range product    412551              1.3%
Not enough data        Not enough data     30200              0.1%


# 09 Export

In [43]:
print(df_ords_prods_merge.columns.tolist())

['order_id', 'product_id', 'add_to_cart_order', 'reordered', 'user_id', 'order_number', 'orders_day_of_week', 'order_hour_of_day', 'days_since_prior_order', 'first_order', 'product_name', 'aisle_id', 'department_id', 'prices', '_merge', 'price_range', 'busiest_day', 'busiest_period_of_day', 'max_order', 'loyalty_flag', 'order_frequency_flag']


In [45]:
# Export enriched dataset as a TEMP version. This will serve as a backup to this point of work incase something goes wrong.
df_ords_prods_merge.to_pickle(os.path.join(
    path,
    '02 Data',
    'Prepared Data',
    'ords_prods_merge_enriched_v1.pkl'
))

In [46]:
# Pickle Export work to main dataset to continue my work
df_ords_prods_merge.to_pickle(os.path.join(path, '02 Data', 'Prepared Data', 'ords_prods_merge.pkl'))

# 10 Rebuilding Dept #/Name list with Product Order Counts

## Imports

In [47]:
# Pickle Export
df_ords_prods_merge.to_pickle(os.path.join(path, '02 Data', 'Prepared Data', 'clean_merge_busiest_day.pkl'))

#### Import Check

In [48]:
df_departments_wrangled = pd.read_csv(os.path.join(path, '02 Data','Prepared Data', 'departments_wrangled.csv'))

In [49]:
# work check
print(df_departments_wrangled.dtypes)
print(df_departments_wrangled.head())

department_id    float64
department        object
product_count      int64
dtype: object
   department_id  department  product_count
0            4.0     produce        9479291
1           16.0  dairy eggs        5398747
2           19.0      snacks        2887550
3            7.0   beverages        2688123
4            1.0      frozen        2234743


In [50]:
# check the number of depts
print(len(df_departments_wrangled))

21


In [51]:
print(df_ords_prods_merge.columns)

Index(['order_id', 'product_id', 'add_to_cart_order', 'reordered', 'user_id',
       'order_number', 'orders_day_of_week', 'order_hour_of_day',
       'days_since_prior_order', 'first_order', 'product_name', 'aisle_id',
       'department_id', 'prices', '_merge', 'price_range', 'busiest_day',
       'busiest_period_of_day', 'max_order', 'loyalty_flag',
       'order_frequency_flag'],
      dtype='object')


## Counts

In [68]:
print(df_counts_named.columns.tolist())

['department_id', 'department', 'product_count']


In [74]:
# Drop the duplicate from the department file (you only need the count from df_counts)
df_counts_named = df_counts_named.drop(columns=['product_count_y'])

# Rename the correct one
df_counts_named = df_counts_named.rename(columns={'product_count_x': 'product_count'})

# Reorder columns
df_counts_named = df_counts_named[['department_id', 'department', 'product_count']]

KeyError: "['product_count_y'] not found in axis"

In [76]:
# Work check to rename/ffix headers
print(df_counts_named.columns.tolist())

['department_id', 'department', 'product_count']


In [77]:
# Counts
df_counts = df_ords_prods_merge['department_id'].value_counts().reset_index()

In [78]:
# rename columns
df_counts.columns = ['department_id', 'product_count']

In [79]:
# work check
print(df_counts)

    department_id  product_count
0             4.0        9479291
1            16.0        5398747
2            19.0        2887550
3             7.0        2688123
4             1.0        2234743
5            13.0        1875369
6             3.0        1172428
7            15.0        1068058
8            20.0        1051249
9             9.0         866627
10           17.0         738666
11           12.0         708927
12           14.0         703033
13           11.0         447572
14           18.0         423802
15            6.0         269253
16            5.0         153696
17            8.0          97716
18           21.0          69145
19            2.0          36291
20           10.0          34573


## Merge with Dept Names

In [80]:
# Re-merge and clean up duplicate columns
df_counts_named = df_counts.merge(df_departments_wrangled, on='department_id', how='left')

# Drop the duplicate from the department file
df_counts_named = df_counts_named.drop(columns=['product_count_y'])

# Rename the correct one
df_counts_named = df_counts_named.rename(columns={'product_count_x': 'product_count'})

# Reorder columns
df_counts_named = df_counts_named[['department_id', 'department', 'product_count']]

In [81]:
print(df_counts_named.columns.tolist())

['department_id', 'department', 'product_count']


In [82]:
# rename and reorder the columns together
df_counts_named = df_counts_named[['department_id', 'department', 'product_count']]

In [83]:
# work check for 
print(df_counts_named)

    department_id       department  product_count
0             4.0          produce        9479291
1            16.0       dairy eggs        5398747
2            19.0           snacks        2887550
3             7.0        beverages        2688123
4             1.0           frozen        2234743
5            13.0           pantry        1875369
6             3.0           bakery        1172428
7            15.0     canned goods        1068058
8            20.0             deli        1051249
9             9.0  dry goods pasta         866627
10           17.0        household         738666
11           12.0     meat seafood         708927
12           14.0        breakfast         703033
13           11.0    personal care         447572
14           18.0           babies         423802
15            6.0    international         269253
16            5.0          alcohol         153696
17            8.0             pets          97716
18           21.0          missing          69145


## Export Data Wrangled

In [84]:
# Export this work
df_counts_named.to_csv(os.path.join(path, '02 Data', 'Prepared Data', 'departments_wrangled.csv'), index=False)